nflow-engine/src/main/java/com/nitorcreations/nflow/engine/internal/executor/WorkflowStateProcessorFactory.java

```diff
@@ -23,6 +34,8 @@
   private final Environment env;
   @Autowired(required = false)
   protected WorkflowExecutorListener[] listeners = new WorkflowExecutorListener[0];
+  final Map<Integer, DateTime> processingInstances = synchronizedMap(new HashMap<Integer, DateTime>());
```

Comment:
With linkedhashmap order where stuck threads are logged ( https://github.com/NitorCreations/nflow/pull/184/commits/f7b3e030fdb0f72a946386d18ea4a03132f510b5#diff-ec7adb4d50d2123d1c7a446126265b75R65 ) would stay the same. But ConcurrentHashMap would be better.